### PR TITLE
[docs] update hello dagster to use deps

### PR DESCRIPTION
**Pull request by jamiedemaria**
## Summary & Motivation

updates the hello dagster project to use `deps` rather than I/O managers

### CODE REVIEW

**Comment by jamiedemaria** [comment]
Current dependencies on/for this PR:
* master
  * **PR #14886** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14886" target="_blank">Graphite</a> 
    * **PR #14888** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14888" target="_blank">Graphite</a> 
      * **PR #14958** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14958" target="_blank">Graphite</a> 
        * **PR #14959** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14959" target="_blank">Graphite</a> 
          * **PR #15052** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15052" target="_blank">Graphite</a> 
            * **PR #15056** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15056" target="_blank">Graphite</a> 
              * **PR #15059** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15059" target="_blank">Graphite</a> 
                * **PR #15327** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15327" target="_blank">Graphite</a>  👈
      * **PR #14887** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14887" target="_blank">Graphite</a> 
        * **PR #14912** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14912" target="_blank">Graphite</a> 
        * **PR #14908** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14908" target="_blank">Graphite</a> 
        * **PR #14896** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14896" target="_blank">Graphite</a> 
        * **PR #14889** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14889" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/15327?utm_source=stack-comment).

**Comment by jamiedemaria** [comment]
Current dependencies on/for this PR:
* master
  * **PR #14886** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14886" target="_blank">Graphite</a> 
    * **PR #14888** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14888" target="_blank">Graphite</a> 
      * **PR #14958** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14958" target="_blank">Graphite</a> 
        * **PR #14959** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14959" target="_blank">Graphite</a> 
          * **PR #15052** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15052" target="_blank">Graphite</a> 
            * **PR #15056** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15056" target="_blank">Graphite</a> 
              * **PR #15059** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15059" target="_blank">Graphite</a> 
                * **PR #15327** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15327" target="_blank">Graphite</a>  👈
      * **PR #14887** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14887" target="_blank">Graphite</a> 
        * **PR #14912** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14912" target="_blank">Graphite</a> 
        * **PR #14908** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14908" target="_blank">Graphite</a> 
        * **PR #14896** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14896" target="_blank">Graphite</a> 
        * **PR #14889** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/14889" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/15327?utm_source=stack-comment).

File: docs/content/getting-started/hello-dagster.mdx [hunk 1]
```diff
@@ -33,13 +35,18 @@ def hackernews_top_story_ids():
...
+
+    with open("hackernews_top_story_ids.json", "w") as f:
+        json.dump(top_story_ids[:10], f)
 
 
 # asset dependencies can be inferred from parameter names
-@asset
-def hackernews_top_stories(hackernews_top_story_ids):
+@asset(deps=[hackernews_top_story_ids])
+def hackernews_top_stories(context):
```
Comment (yuhan) [inline docs/content/getting-started/hello-dagster.mdx:45]: can we add type here

File: examples/docs_snippets/docs_snippets/getting-started/hello-dagster/hello-dagster.py [hunk 2]
```diff
@@ -13,13 +15,18 @@ def hackernews_top_story_ids():
...
+
+    with open("hackernews_top_story_ids.json", "w") as f:
+        json.dump(top_story_ids[:10], f)
 
 
 # asset dependencies can be inferred from parameter names
-@asset
-def hackernews_top_stories(hackernews_top_story_ids):
+@asset(deps=[hackernews_top_story_ids])
+def hackernews_top_stories(context):
```
Comment (yuhan) [inline examples/docs_snippets/docs_snippets/getting-started/hello-dagster/hello-dagster.py:25]: type too

**Review by yuhan — APPROVED** [review]

**Pull request merged**